lib/chat_review_queue.rb

```diff
@@ -117,4 +124,29 @@ def companion_pm_creator(chat_message, flagger, flag_type_id, opts)
 
     PostCreator.new(flagger, create_args)
   end
+
+  def can_flag_again?(reviewable, message, flagger, flag_type_id)
+    return true if reviewable.blank?
+
+    flagger_has_no_pending_flags =
+      reviewable.reviewable_scores.none? { |rs| rs.user == flagger && rs.pending? }
+
+    if flagger_has_no_pending_flags && flag_type_id == PostActionType.types[:notify_moderators]
```

Comment:
I don't see any test using this :notify_moderators

Reply:
From the PR description, it says `“A reviewable can only have one pending flag of each type, except for notify_moderators”` so I guess here we should use `||` instead of `&&`, right?

Reply:
Since `notify_moderators`is always available, we need to make sure the user has no other pending flags. You can only flag a message once.